qiita_db/support_files/patches/python_patches/14.py

```diff
@@ -0,0 +1,20 @@
+# Feberuary 7, 2015
+# This patch recreates all the QIIME mapping files to avoid lower-upper case
+# problems. See https://github.com/biocore/qiita/issues/799
+
+from qiita_db.util import get_mountpoint
+from qiita_db.sql_connection import SQLConnectionHandler
+from qiita_db.metadata_template import PrepTemplate
+
+conn_handler = SQLConnectionHandler()
+
+_id, fp_base = get_mountpoint('templates')[0]
+
+for prep_template_id in conn_handler.execute_fetchall(
+        "SELECT prep_template_id FROM qiita.prep_template"):
+    prep_template_id = prep_template_id[0]
+    pt = PrepTemplate(prep_template_id)
+    study_id = pt.study_id
+
+    for _, fpt in pt.get_filepaths():
+        pt.create_qiime_mapping_file(fpt)
```

Comment:
So just to be sure, if I do that, the code will not be deleting or
forgetting the old/incorrect QIIME mapping file, it will instead just
add one more file to the prep template? If that's the case then I think
we need to remove the old mapping files because of this few lines of
code:
https://github.com/biocore/qiita/blob/7b0630dc7ddca95d38f9f5ec026221b6e6dc254e/qiita_pet/uimodules/raw_data_tab.py#L219-225

On (Feb-09-15|10:09), adamrp wrote:
> > +from qiita_db.util import get_mountpoint
> > +from qiita_db.sql_connection import SQLConnectionHandler
> > +from qiita_db.metadata_template import PrepTemplate
> > +
> > +conn_handler = SQLConnectionHandler()
> > +
> > +_id, fp_base = get_mountpoint('templates')[0]
> > +
> > +for prep_template_id in conn_handler.execute_fetchall(
> > +        "SELECT prep_template_id FROM qiita.prep_template"):
> > +    prep_template_id = prep_template_id[0]
> > +    pt = PrepTemplate(prep_template_id)
> > +    study_id = pt.study_id
> > +
> > +    for _, fpt in pt.get_filepaths():
> > +        pt.create_qiime_mapping_file(fpt)
> 
> Good question. I think that theoretically, it should be okay to do it this way, but I'm slightly hesitant because I'm not sure if there are places in the code that unadvisable assumptions are made with regard to the number of templates associated with a study (e.g., is there code that assumes that there is only one prep template, and if so, does that code presume to use the prep template with the highest ID, or the lowest ID?)
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/829/files#r24349990